paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -221,13 +218,16 @@ protected boolean databaseExistsImpl(String databaseName) {
     @Override
     protected void createDatabaseImpl(String name) {
         try {
-            Path databasePath = newDatabasePath(name);
-            locationHelper.createPathIfRequired(databasePath, fileIO);
-
-            Database database = new Database();
-            database.setName(name);
-            locationHelper.specifyDatabaseLocation(databasePath, database);
-            client.createDatabase(database);
+            Database database = getDatabase(name);
+            if (database == null) {
```

Comment:
If database not null, here should throw `DatabaseAlreadyExistException`?
We can add test in `CatalogTestBase`.

Reply:
No need to add   Database database = getDatabase(name);  if (database == null) {...} here,

see 

```java
    public void createDatabase(String name, boolean ignoreIfExists, Map<String, String> properties)
            throws DatabaseAlreadyExistException {
        if (isSystemDatabase(name)) {
            throw new ProcessSystemDatabaseException();
        }
        if (databaseExists(name)) {
            if (ignoreIfExists) {
                return;
            }
            throw new DatabaseAlreadyExistException(name);
        }
        createDatabaseImpl(name, properties);
    }
```
user should use createDatabase instead of createDatabaseImpl

Reply:
this pr can be revert actually